### ISSUE TITLE
Some output being dropped since upgrading to `through2`

### ISSUE THREAD

**Issue by dfreeman** [ember-cli/console-ui#67]
This was originally reported by @josemarluedke when he noticed TypeScript errors not being emitted, but after a little more investigation it looks like other output is missing as well.

In [@josemarluedke's reproduction repo](https://github.com/josemarluedke/ts-no-errors), running `ember build` only emits the following:

```
WARNING: ts-no-errors has added the class-properties plugin to its build, but ember-cli-babel provides these by default now! You can remove the transforms, or the addon that provided them, such as @ember-decorators/babel-transforms.
```

I can also briefly see the `Building...` spinner and `cleaning up` message, but those are quickly overwritten.

If I set a resolution for `console-ui` to `3.0.0` and rerun `yarn`, I see the full output I'd expect:

```
WARNING: ts-no-errors has added the class-properties plugin to its build, but ember-cli-babel provides these by default now! You can remove the transforms, or the addon that provided them, such as @ember-decorators/babel-transforms.
WARNING: ts-no-errors has added the class-properties plugin to its build, but ember-cli-babel provides these by default now! You can remove the transforms, or the addon that provided them, such as @ember-decorators/babel-transforms.
Environment: development
app/components/navigation.ts:4:3 - error TS2564: Property 'someArg' has no initializer and is not definitely assigned in the constructor.

4   someArg: boolean;
    ~~~~~~~

app/components/navigation.ts:6:7 - error TS2378: A 'get' accessor must return a value.

6   get shouldGiveAnError() {
        ~~~~~~~~~~~~~~~~~

app/components/navigation.ts:7:5 - error TS2349: Cannot invoke an expression whose type lacks a call signature. Type 'Boolean' has no compatible call signatures.

7     this.someArg();
      ~~~~~~~~~~~~~~

tests/integration/components/navigation-test.ts:15:18 - error TS2531: Object is possibly 'null'.

15     assert.equal(this.element.textContent.trim(), '');
                    ~~~~~~~~~~~~~~~~~~~~~~~~

tests/integration/components/navigation-test.ts:24:18 - error TS2531: Object is possibly 'null'.

24     assert.equal(this.element.textContent.trim(), 'template block text');
                    ~~~~~~~~~~~~~~~~~~~~~~~~

cleaning up...
Built project successfully. Stored in "dist/".
```

**Comment by stefanpenner** [comment]
Cc @gabrielcsapo looks like something may have gone sideways with through2

We may want to revert for now, and when time permits figure out what’s up, write tests and fix what’s going on.

**Comment by stefanpenner** [comment]
Might be semi related: https://github.com/ember-cli/console-ui/issues/60

**Comment by gabrielcsapo** [comment]
Looking into this now 👨‍💻

**Comment by rwjblue** [comment]
Awesome, thanks for jumping on it quickly.

@igorT - This is what you and I noticed, while pairing earlier.

**Comment by gabrielcsapo** [comment]
Found the issue, writing a test to ensure this doesn't happen again

**Comment by stefanpenner** [comment]
should be fixed with: v3.0.2 🎉

thanks @gabrielcsapo  for the quick fix!
